libsycl/src/CMakeLists.txt

```diff
@@ -49,7 +47,7 @@ function(add_sycl_rt_library LIB_TARGET_NAME LIB_OBJ_NAME LIB_OUTPUT_NAME)
     target_compile_options(${LIB_OBJ_NAME} PUBLIC /EHsc)
   else()
     target_compile_options(${LIB_OBJ_NAME} PUBLIC
-                           -fvisibility=hidden -fvisibility-inlines-hidden)
+                           -fvisibility=hidden -fvisibility-inlines-hidden -fPIC)
```

Comment:
Instead of passing `-fPIC` explicitly, set the `POSITION_INDEPENDENT_CODE` property to `ON` via `set_target_properties`. This is what liboffload does (see `offload/liboffload/CMakeLists.txt`).